presidio-analyzer/analyzer/entity_recognizer.py

```diff
@@ -86,3 +94,149 @@ def to_dict(self):
+            self.logger.debug('match is \'%s\'', text[result.start:result.end])
+            context = self.__extract_context(
+                nlp_artifacts=nlp_artifacts,
+                word=text[result.start:result.end],
+                start=result.start)
+
+            self.logger.debug('context is %s', context)
+            context_similarity = self.__calculate_context_similarity(
+                context, predefined_context_words)
+            if context_similarity >= \
+               self.CONTEXT_SIMILARITY_THRESHOLD:
+                result.score += \
+                  context_similarity * self.CONTEXT_SIMILARITY_FACTOR
+                result.score = max(
+                    result.score,
+                    self.MIN_SCORE_WITH_CONTEXT_SIMILARITY)
+                result.score = min(
+                    result.score,
+                    EntityRecognizer.MAX_SCORE)
+        return results
+
+    @staticmethod
+    def __context_to_keywords(context):
+        return context.split(' ')
+
+    def __calculate_context_similarity(self,
+                                       context_text,
+                                       context_list):
+        """Context similarity is 1 if there's exact match between a keyword in
+           context_text and any keyword in context_list
+
+        :param context_text a string of the prefix and suffix of the found
+               match
+        :param context_list a list of words considered as context keywords
+        """
+
+        if context_list is None:
+            return 0
+
+        context_keywords = self.__context_to_keywords(context_text)
+        if context_keywords is None:
+            return 0
+
+        similarity = 0.0
+        for context_keyword in context_keywords:
+            if context_keyword in context_list:
+                self.logger.info("Found context keyword '%s'", context_keyword)
+                similarity = 1
+                break
+
+        return similarity
+
+    def __extract_context(self, nlp_artifacts, word, start):
+        """ Extracts words surronding another given word.
+            The text from which the context is extracted is given in the nlp
+            doc
+            :param nlp_artifacts: An abstraction layer which holds different
+                                  items which are result of a NLP pipeline
+                                  execution on a given text
+            :param word: The word to look for context around
+            :param start: The start index of the word in the original text
+        """
+
+        if not nlp_artifacts.tokens:
+            self.logger.info('Skipping context around \'%s\'', word)
+            # if there are no nlp artifacts, this is ok, we can
+            # extract context and we return a valid, yet empty
+            # context
+            return ''
+
+        self.logger.debug('Extracting context around \'%s\'', word)
+        context_keywords = nlp_artifacts.keywords
```

Comment:
replace with lemmatized_keywords or something like that

Reply:
👍